src/main/scala/org/neo4j/spark/service/Neo4jQueryService.scala

```diff
@@ -1,39 +1,77 @@
-import org.neo4j.cypherdsl.core.{Condition, Conditions, Cypher, Functions, PropertyContainer, StatementBuilder}
+import org.apache.spark.sql.sources.{And, Filter, Or}
 import org.neo4j.cypherdsl.core.renderer.Renderer
-import org.neo4j.spark.{Neo4jOptions, QueryType}
+import org.neo4j.cypherdsl.core._
 import org.neo4j.spark.util.Neo4jImplicits._
 import org.neo4j.spark.util.Neo4jUtil
+import org.neo4j.spark.{Neo4jOptions, NodeSaveMode, QueryType}
 
-import collection.JavaConverters._
+import scala.collection.JavaConverters._
 
 class Neo4jQueryWriteStrategy(private val saveMode: SaveMode) extends Neo4jQueryStrategy {
   override def createStatementForQuery(options: Neo4jOptions): String =
-    s"""UNWIND ${"$"}events AS event
+    s"""UNWIND ${"$"}events AS $BATCH_VARIABLE
        |${options.query.value}
        |""".stripMargin
 
-  override def createStatementForRelationships(options: Neo4jOptions): String = throw new UnsupportedOperationException("TODO implement method")
+  private def keywordFromSaveMode(saveMode: Any): String = {
+    saveMode match {
+      case NodeSaveMode.Overwrite | SaveMode.Overwrite => "MERGE"
+      case NodeSaveMode.ErrorIfExists | SaveMode.ErrorIfExists => "CREATE"
+      case NodeSaveMode.Match => "MATCH"
+      case _ => throw new UnsupportedOperationException(s"Source SaveMode $saveMode not supported")
+    }
+  }
+
+  override def createStatementForRelationships(options: Neo4jOptions): String = {
+    val relationshipKeyword = keywordFromSaveMode(saveMode)
+    val sourceKeyword = keywordFromSaveMode(options.relationshipMetadata.sourceSaveMode)
+    val targetKeyword = keywordFromSaveMode(options.relationshipMetadata.targetSaveMode)
+
+    val relationship = options.relationshipMetadata.relationshipType.quote()
+
+    val sourceLabels = options.relationshipMetadata.source.labels
+      .map(_.quote())
+      .mkString(":")
+
+    val targetLabels = options.relationshipMetadata.target.labels
+      .map(_.quote())
+      .mkString(":")
+
+    val sourceKeys = options.relationshipMetadata.source.nodeKeys.map(key => {
+      s"${key._2.quote()}:$BATCH_VARIABLE.source.${Neo4jWriteMappingStrategy.KEYS}.${key._1.removeAlias().quote()}"
+    }).mkString(", ")
+    val targetKeys = options.relationshipMetadata.target.nodeKeys.map(key => {
+      s"${key._2.quote()}:$BATCH_VARIABLE.target.${Neo4jWriteMappingStrategy.KEYS}.${key._1.removeAlias().quote()}"
+    }).mkString(", ")
+
+    val sourceQueryPart = createQueryPart(sourceKeyword, sourceLabels, sourceKeys, Neo4jUtil.RELATIONSHIP_SOURCE_ALIAS)
+    val targetQueryPart = createQueryPart(targetKeyword, targetLabels, targetKeys, Neo4jUtil.RELATIONSHIP_TARGET_ALIAS, Seq(Neo4jUtil.RELATIONSHIP_SOURCE_ALIAS))
+
+    s"""UNWIND ${"$"}events AS $BATCH_VARIABLE
+       |$sourceQueryPart
+       |$targetQueryPart
+       |$relationshipKeyword (${Neo4jUtil.RELATIONSHIP_SOURCE_ALIAS})-[${Neo4jUtil.RELATIONSHIP_ALIAS}:${relationship}]->(${Neo4jUtil.RELATIONSHIP_TARGET_ALIAS})
```

Comment:
damn that's a good catch (☞ﾟヮﾟ)☞

Reply:
Table-flipping emoji guy is my go-to.  He exists in the unit tests for most code I write, because he's really good at fucking things up.   The backticks are just icing on the cake that happens to do the same with Cypher.